### PR TITLE
libckteec: Add EDDSA attribute serialization

### PR DESCRIPTION
**Pull request by varder**
The PKCS#11 Specification:
https://docs.oasis-open.org/pkcs11/pkcs11-spec/v3.1/cs01/pkcs11-spec-v3.1-cs01.pdf

The pull request should be merged with the following pull requests:
https://github.com/OP-TEE/optee_os/pull/5559
https://github.com/OP-TEE/optee_test/pull/618

### CODE REVIEW

File: libckteec/include/pkcs11.h [hunk 1]
```diff
@@ -230,6 +230,7 @@ typedef CK_KEY_TYPE *CK_KEY_TYPE_PTR;
 #define CKK_DH				0x002
 #define CKK_ECDSA			0x003
 #define CKK_EC				0x003
+#define CKK_EC_EDWARDS		0x040
```
Comment (etienne-lms) [inline libckteec/include/pkcs11.h:233]: to move at the end (numerical order)

I propose you mention that this is from pkcs11 v3.1-cs01:
`#define CKK_EC_EDWARDS		0x040 /* from PKCS#11 v3.1-cs01 */

File: libckteec/include/pkcs11.h [hunk 2]
```diff
@@ -350,6 +352,7 @@ typedef CK_MECHANISM_TYPE *CK_MECHANISM_TYPE_PTR;
 #define CKM_ECDH1_COFACTOR_DERIVE	0x01051
 #define CKM_ECMQV_DERIVE		0x01052
 #define CKM_ECDH_AES_KEY_WRAP		0x01053
+#define CKM_EDDSA				0x01057
```
Comment (etienne-lms) [inline libckteec/include/pkcs11.h:355]:
```patch
 #define CKM_ECMQV_DERIVE		0x01052
 #define CKM_ECDH_AES_KEY_WRAP		0x01053
 #define CKM_RSA_AES_KEY_WRAP		0x01054
+#define CKM_EC_EDWARDS_KEY_PAIR_GEN	0x01055
+#define CKM_EDDSA			0x01057
 #define CKM_AES_KEY_GEN			0x01080
 #define CKM_AES_ECB			0x01081

File: libckteec/include/pkcs11_ta.h [hunk 3]
```diff
@@ -1276,6 +1276,7 @@ enum pkcs11_mechanism_id {
 	PKCS11_CKM_ECDSA_SHA512			= 0x01046,
 	PKCS11_CKM_ECDH1_DERIVE			= 0x01050,
 	PKCS11_CKM_ECDH1_COFACTOR_DERIVE	= 0x01051,
+	PKCS11_CKM_EDDSA			= 0x01057,
```
Comment (etienne-lms) [inline libckteec/include/pkcs11_ta.h:1279]: move below

**Review by etienne-lms — COMMENTED** [review]
LGTM aside these few comments.

File: libckteec/include/pkcs11.h (hunk 2 above)
Comment (varder) [reply libckteec/include/pkcs11.h]: fixed

File: libckteec/include/pkcs11.h [hunk 4]
```diff
@@ -492,6 +495,15 @@ struct CK_GCM_PARAMS {
 	CK_ULONG		ulTagBits;
 };
 
+/* EDDSA */
```
[the remark this answers is not in the document]
Comment (varder) [reply libckteec/include/pkcs11.h]: done

File: libckteec/include/pkcs11.h [hunk 5]
```diff
@@ -351,6 +352,8 @@ typedef CK_MECHANISM_TYPE *CK_MECHANISM_TYPE_PTR;
 #define CKM_ECMQV_DERIVE		0x01052
 #define CKM_ECDH_AES_KEY_WRAP		0x01053
 #define CKM_RSA_AES_KEY_WRAP		0x01054
+#define CKM_EC_EDWARDS_KEY_PAIR_GEN	0x01055
+#define CKM_EDDSA					0x01057
```
Comment (etienne-lms) [inline libckteec/include/pkcs11.h:356]: nitpicking: remove 1 or 2 tabulations

File: libckteec/include/pkcs11_ta.h [hunk 6]
```diff
@@ -1279,6 +1279,7 @@ enum pkcs11_mechanism_id {
 	PKCS11_CKM_ECMQV_DERIVE			= 0x01052,
 	PKCS11_CKM_ECDH_AES_KEY_WRAP		= 0x01053,
 	PKCS11_CKM_RSA_AES_KEY_WRAP		= 0x01054,
+	PKCS11_CKM_EDDSA				= 0x01057,
```
Comment (etienne-lms) [inline libckteec/include/pkcs11_ta.h:1282]: remove 1 tab

**Review by etienne-lms — COMMENTED** [review]
`Reviewed-by: Etienne Carriere <etienne.carriere@linaro.org>` with 2 minor comments addressed.

File: libckteec/include/pkcs11.h (hunk 5 above)
Comment (varder) [reply libckteec/include/pkcs11.h]: fixed

File: libckteec/include/pkcs11_ta.h (hunk 6 above)
Comment (varder) [reply libckteec/include/pkcs11_ta.h]: fixed

**Pull request merged**